localstack/services/awslambda/lambda_executors.py

```diff
@@ -854,8 +904,13 @@ def prime_docker_container(
 
                 LOG.debug("Starting docker-reuse Lambda container: %s", container_name)
                 DOCKER_CLIENT.start_container(container_name)
+
+                def wait_up():
+                    cont_status = DOCKER_CLIENT.get_container_status(container_name)
```

Comment:
We wait here for the container to start up, but would that necessarily include that the 9001 socket is already available? It only checks if the container status is up, not whether the port is already open. Might be a small window we never hit though.
This would also tie in nicely with the availability check for port 9001 to check against networking issues.

Reply:
Great point, thanks @dfangl . 👍  Updated the PR - it now also contains logic to assert availability of the target port, where appropriate (only in case we're running in Docker, and if the IP address of the target container can be determined by the Docker client).